sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -450,6 +456,10 @@ private static String buildQuery(Read spec) {
               .map(ColumnMetadata::getName)
               .collect(Collectors.joining(","));
 
+      List<TokenRange> tokenRanges =
+          getTokenRanges(cluster, spec.keyspace().get(), spec.table().get());
+      final long estimatedSize = getEstimatedSizeBytesFromTokenRanges(tokenRanges);
```

Comment:
That's good point. Based on past implementation I assumed very bad estimation but I didn't want to change this logic. 
What I could try to do is based on root estimates (SA) and n splits calculate SA_1 = SA/n etc. Generally this size estimates is also broken from beginning as CassandraIO can query 1 column instead of all or have filters and this estimation doesn't take it into account. It supports only size(t) where query is select * from table t. In A_1,A_2.. it's difficult to calculate size as ranges are concatenated to underlying queries.

Reply:
fixed this logic.